File: leetcode/12.cs
public class Solution {
    public string IntToRoman(int num) {
        if(num / 1000 > 0){
            return "M" + this.IntToRoman(num - 1000);
        }
        if(num / 100 > 0){
            if(num /100 == 9){
               return "CM" + this.IntToRoman(num - 900);
            }
            if(num /100 == 4){
               return "CD" + this.IntToRoman(num - 400);
            }
            if(num/500 >= 1){
                return "D" + this.IntToRoman(num - 500);
            }
            return "C" + this.IntToRoman(num - 100);
        }
        if(num / 10 > 0){
            if(num /10 == 9){
               return "XC" + this.IntToRoman(num - 90);
            }
            if(num /10 == 4){
               return "XL" + this.IntToRoman(num - 40);
            }
            if(num/50 >= 1){
                return "L" + this.IntToRoman(num - 50);
            }
            else{
                return "X" + this.IntToRoman(num - 10);
            }
        }
        if(num > 0){
            if(num == 9){
               return "IX";
            }
            if(num == 4){
               return "IV";
            }
            if(num/5 >= 1){
                return "V" + this.IntToRoman(num - 5);
            }
            return "I" + this.IntToRoman(num - 1);
        }
        else{
            return "";
        }
    }
}

File: leetcode/12.ts
function intToRoman(num: number): string {
    if(Math.floor(num/1) <= 0){
        return "";
    }
    else if(Math.floor(num/1000) > 0){
        return "M" + intToRoman(num - 1000);
    }
    else if(Math.floor(num/900) > 0){
        return "CM" + intToRoman(num - 900);
    }
    else if(Math.floor(num/500) > 0){
        return "D" + intToRoman(num - 500);
    }
    else if(Math.floor(num/400) > 0){
        return "CD" + intToRoman(num - 400);
    }
    else if(Math.floor(num/100) > 0){
        return "C" + intToRoman(num - 100);
    }
    else if(Math.floor(num/90) > 0){
        return "XC" + intToRoman(num - 90);
    }
    else if(Math.floor(num/50) > 0){
        return "L" + intToRoman(num - 50);
    }
    else if(Math.floor(num/40) > 0){
        return "XL" + intToRoman(num - 40);
    }
    else if(Math.floor(num/10) > 0){
        return "X" + intToRoman(num - 10);
    }
    else if(Math.floor(num/9) > 0){
        return "IX" + intToRoman(num - 9);
    }
    else if(Math.floor(num/5) > 0){
        return "V" + intToRoman(num - 5);
    }
    else if(Math.floor(num/4) > 0){
        return "IV" + intToRoman(num - 4);
    }
    else {
        return "I" + intToRoman(num - 1);
    }
};
